spec/lib/migrations/load_rate_reference_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+require 'rake'
+require 'roo'
+
+RSpec.describe 'Load Rating Regions Task', :type => :task do
+  
+  context "rate_reference:update_rating_regions" do
+    before :all do
+      Rake.application.rake_require "tasks/migrations/load_rate_reference"
+      Rake::Task.define_task(:environment)
+    end
+    it 'expect not raise errors' do
+      expect { invoke_task }.not_to raise_error
+    end
+    
+    it 'expect to read xlsx file' do
+      file_path = File.join(Rails.root,'lib', 'xls_templates', "SHOP_ZipCode_CY2017_FINAL.xlsx")
+      xlsx = Roo::Spreadsheet.open(file_path)
+      expect(xlsx.sheet(0).row(2)).to match_array(["01001", "Hampden", "No", "Rating Area 1"])
+      expect(xlsx.sheet(0).row(3)).to match_array(["01002", "Hampshire", "Yes", "Rating Area 1"])
+      expect(xlsx.sheet(0).row(4)).to match_array(["01002", "Franklin", "Yes", "Rating Area 1"])
+      expect(xlsx.sheet(0).row(699)).to match_array(["02791", "Bristol", "No", "Rating Area 6"])
+    end
+    
+    it "should have the right data" do
+      file_path = File.join(Rails.root,'lib', 'xls_templates', "SHOP_ZipCode_CY2017_FINAL.xlsx")
+      xlsx = Roo::Spreadsheet.open(file_path)
+      expect(xlsx.sheet(0).row(2).first.to_i).to eq RateReference.first.zip_code
+      expect(xlsx.sheet(0).row(2)[1]).to eq RateReference.first.county
+      expect(xlsx.sheet(0).row(2).last).to eq RateReference.first.rating_region
+    end
+
+    it "should not have duplicate data" do
+      file_path = File.join(Rails.root,'lib', 'xls_templates', "SHOP_ZipCode_CY2017_FINAL.xlsx")
+      xlsx = Roo::Spreadsheet.open(file_path)
+      sheet = xlsx.sheet(0)
+      rate_references = RateReference.where(zip_code: sheet.cell(2,1), county: sheet.cell(2,2), multiple_counties: to_boolean(sheet.cell(2,3)),
+        rating_region: sheet.cell(2,4))
+      expect(rate_references.size).to eq 1
+    end
+    
+    private
+    
+    def invoke_task
+      Rake::Task["load_rate_reference:update_rating_regions"].invoke
+    end
+    
+    def to_boolean(value)
```

Comment:
This is another symptom of the duplication / tight coupling between this test and the thing it's testing